### ISSUE TITLE
quickFilters Filter assistant mode is not detected when configured to automatically start with Thunderbird

### ISSUE THREAD

**Issue by PotatoCarl** [RealRaven2000/QuickFolders#397]
Hi, when using QF 6.0pre332 and QuickFilter 6.1.1 with TB115 and I move a message via the QuickMove button of QuickFolder to a folder, the Filter dialog is not opening (and no filter is created).
When moving directly to a filder in the folder view of TB, it works.

**Comment by RealRaven2000** [comment]
Thanks for reporting, can you please check the error console log (open with Ctrl+Shift+J or Thunderbird tools » ) to see if there are any new errors logged _after_ selecting the target folder on QuickFolders?

You can copy the debug log (as text please, no screenshot) via _right-click » Copy all Messages_ :

![image](https://github.com/RealRaven2000/quickFilters/assets/6314674/aea6a33f-8008-448f-a038-f69ba19b1825)

Error console settings should be set up like this to be most useful:

![image](https://github.com/RealRaven2000/quickFilters/assets/6314674/a26a7586-290d-4137-92ac-4ae1e1323237)

I can give you additional debug settings for quickFilters for more detail later, which will be configured on _quickFilters Settings » Advanced_, but let's first see if there is anything obvious...

![image](https://github.com/RealRaven2000/quickFilters/assets/6314674/be2788d9-02f3-4496-bc8a-553b01c88d12)

**Comment by PotatoCarl** [comment]
QuickFolders 12:22:52.504  [11016 ms]   
 dragSession =  
XPCWrappedNative_NoHelper { QueryInterface: QueryInterface(), invokeDragSession: invokeDragSession(), getCurrentSession: getCurrentSession(), startDragSession: startDragSession(), startDragSessionForTests: startDragSessionForTests(), endDragSession: endDragSession(), suppress: suppress(), unsuppress: unsuppress(), DRAGDROP_ACTION_NONE: 0, DRAGDROP_ACTION_COPY: 1, … }
quickfolders-util.js:1286:13
QuickFolders 12:22:52.506  [2 ms]   
 toggleMoveModeSearchBox(true) quickfolders-util.js:1286:13
QuickFolders 12:22:52.506  [0 ms]   
 QuickFolders.quickMove.update()
isActive = true quickfolders-util.js:1286:13
QuickFolders 12:22:52.507  [1 ms]   
 toggleMoveModeSearchBox(true) quickfolders-util.js:1286:13
QuickFolders 12:22:52.507  [0 ms]   
 toggleMoveModeSearchBox(true) quickfolders-util.js:1286:13
QuickFolders 12:22:53.531  [1024 ms]   
 findFolder (e) quickfolders-util.js:1286:13
QuickFolders 12:22:53.716  [185 ms]   
 findFolder (ec) quickfolders-util.js:1286:13
QuickFolders 12:22:53.832  [116 ms]   
 findFolder (eco) quickfolders-util.js:1286:13
QuickFolders 12:22:54.10  [178 ms]   
 findFolder (ecop) quickfolders-util.js:1286:13
QuickFolders 12:22:54.182  [172 ms]   
 findFolder (ecopo) quickfolders-util.js:1286:13
QuickFolders 12:22:54.314  [132 ms]   
 findFolder (ecopox) quickfolders-util.js:1286:13
QuickFolders 12:22:54.814  [500 ms]   
 findFolder (ecopox) quickfolders-util.js:1286:13
QuickFolders 12:22:55.628  [814 ms]   
 selectFound - [object XULCommandEvent] quickfolders-util.js:1286:13
QuickFolders 12:22:55.629  [1 ms]   
 moveMessages  with readStatus = 0 quickfolders-util.js:1286:13
QuickFolders 12:22:55.652  [23 ms]   
 util.touch(Ecopox)
1690885375
1/8 12:22:55 quickfolders-util.js:1286:13
msgsMoveCopyCompleted()
 
Object { isMoved: true, aSrcMsgs: (1) […], targetFolder: XPCWrappedNative_NoHelper, aDestMsgs: (1) […] }
quickFilters.js:1771:15
quickFilters 12:22:55.654  [568819 ms]   
New assistant mode disabled, msgsMoveCopyCompleted does not invoke filter assistant.
QuickFolders 12:22:55.741  [89 ms]   
 util.touch(Ecopox)
1690885375
1/8 12:22:55 quickfolders-util.js:1286:13
QuickFolders 12:22:55.742  [1 ms]   
 QuickFolders.quickMove.update()
isActive = false quickfolders-util.js:1286:13
QuickFolders 12:22:55.743  [1 ms]   
 toggleMoveModeSearchBox(false) quickfolders-util.js:1286:13
[JavaScript Warning: "findFolder (false, null) failed.
threadPane is null" {file: "chrome://quickfolders/content/quickfolders-interface.js" line: 5491 column: 0 source: "findFolder
  chrome://quickfolders/content/quickfolders-interface.js:5491:9
update@chrome://quickfolders/content/quickfolders-quickMove.js:382:8
resetList@chrome://quickfolders/content/quickfolders-quickMove.js:246:10
execute@chrome://quickfolders/content/quickfolders-quickMove.js:206:9
"}] quickfolders-util.js:1312:15
QuickFolders 12:22:55.744  [1 ms]   
 QuickFolders.quickMove.update()
isActive = false quickfolders-util.js:1286:13
QuickFolders 12:22:55.744  [0 ms]   
 toggleMoveModeSearchBox(false) quickfolders-util.js:1286:13
[JavaScript Warning: "findFolder (false, null) failed.
threadPane is null" {file: "chrome://quickfolders/content/quickfolders-interface.js" line: 5491 column: 0 source: "findFolder
  chrome://quickfolders/content/quickfolders-interface.js:5491:9
update@chrome://quickfolders/content/quickfolders-quickMove.js:382:8
execute@chrome://quickfolders/content/quickfolders-quickMove.js:207:9
"}] quickfolders-util.js:1312:15
QuickFolders 12:22:56.550  [806 ms]   
 util.touch(Ecopox)
1690885376
1/8 12:22:56 quickfolders-util.js:1286:13

**Comment by RealRaven2000** [comment]
thanks for the log. are you in a single message tab?

**Comment by PotatoCarl** [comment]
No so sure what you mean. I have the old 3way view open, folder tree, contents and (selected) email.

**Comment by RealRaven2000** [comment]
> No so sure what you mean. I have the old 3way view open, folder tree, contents and (selected) email.

ok, single message tab happens when you open a mail in a new tab (via double click). There is an error thrown about missing thread pane, which would indicate that the list of messages (on top of the mail preview) is hidden. To avoid the error and test with something that is closer to the latest source code can you please install the version below (6.1 prerelease 34) - the preview you used is behind quite a bit.

[QuickFolders-mx-6.1pre34.zip](https://github.com/RealRaven2000/quickFilters/files/12228705/QuickFolders-mx-6.1pre34.zip)

I will do some more testing in the meantime... however, I don't see any specific error that I can reproduce right now.

**Comment by PotatoCarl** [comment]
No change.
The list of messages is displayed quite well.

**Comment by PotatoCarl** [comment]
Hi
After Upgrade to QuickFilter 6.2, the QuickMove function does not come up with the filter dialog when sorting from a subfolder anymore. Worked with 6.1.1pre11

**Comment by PotatoCarl** [comment]
Sorry does not work when 1st time involved, works with 2. time...

**Comment by PotatoCarl** [comment]
Does not work with 6.2 anymore... Not first, not second time (when moving to the QuickFolder Quickjump button with 6.1.2pre11).
Or Moving with The Shortkey.

**Comment by RealRaven2000** [comment]
> Does not work with 6.2 anymore... Not first, not second time (when moving to the QuickFolder Quickjump button with 6.1.2pre11). Or Moving with The Shortkey.

Sorry can you be more specific, I think what you are saying is - quickFilters 6.2 + QF 6.1.2pre11; you are using the quickMove function and selecting a suggested menu item directly underneath the button? Or a in a child folder from there? 

I am asking again because it works for me. If you get any messages in error console after selecting the folder from the menu, I need to see them please.

**Comment by RealRaven2000** [comment]
Just in case, here is the very latest version of QuickFolders (6.1.2 pre 24):

[QuickFolders-mx-6.1.2pre24.zip](https://github.com/RealRaven2000/quickFilters/files/12395691/QuickFolders-mx-6.1.2pre24.zip)

**Comment by PotatoCarl** [comment]
Nope, does not work with Pre24 either.
I might have the same bug open multiple time, but since the 115 version I have massive problems with the move/filter
Sometimes it works, sometimes it does not.

Here is the log when after move nothing happens (i.e. not filter dialog pops open):
QuickFolders 14:31:11.404  [13296 ms]   
 dragSession =  
XPCWrappedNative_NoHelper { QueryInterface: QueryInterface(), invokeDragSession: invokeDragSession(), getCurrentSession: getCurrentSession(), startDragSession: startDragSession(), startDragSessionForTests: startDragSessionForTests(), endDragSession: endDragSession(), suppress: suppress(), unsuppress: unsuppress(), DRAGDROP_ACTION_NONE: 0, DRAGDROP_ACTION_COPY: 1, … }
quickfolders-util.js:1292:13
QuickFolders 14:31:11.407  [3 ms]   
 toggleMoveModeSearchBox(true) quickfolders-util.js:1292:13
QuickFolders 14:31:11.407  [0 ms]   
 QuickFolders.quickMove.update()
isActive = true quickfolders-util.js:1292:13
QuickFolders 14:31:11.408  [1 ms]   
 toggleMoveModeSearchBox(true) quickfolders-util.js:1292:13
QuickFolders 14:31:11.409  [1 ms]   
 toggleMoveModeSearchBox(true) quickfolders-util.js:1292:13
QuickFolders 14:31:13.708  [2299 ms]   
 findFolder (ärzt) quickfolders-util.js:1292:13
QuickFolders 14:31:13.807  [99 ms]   
 findFolder (ärzte) quickfolders-util.js:1292:13
QuickFolders 14:31:14.307  [500 ms]   
 findFolder (ärzte) quickfolders-util.js:1292:13
QuickFolders 14:31:16.127  [1820 ms]   
 selectFound - [object XULCommandEvent] quickfolders-util.js:1292:13
QuickFolders 14:31:16.128  [1 ms]   
 moveMessages  with readStatus = 0 quickfolders-util.js:1292:13
QuickFolders 14:31:16.139  [11 ms]   
 util.touch(Ärzte)
1692621076
21/8 14:31:16 quickfolders-util.js:1292:13
msgsMoveCopyCompleted()
 
Object { isMoved: true, aSrcMsgs: (1) […], targetFolder: XPCWrappedNative_NoHelper, aDestMsgs: (1) […] }
quickFilters.js:1827:15
quickFilters 14:31:16.142  [153436 ms]   
New assistant mode disabled, msgsMoveCopyCompleted does not invoke filter assistant.
QuickFolders 14:31:16.201  [62 ms]   
 util.touch(Ärzte)
1692621076
21/8 14:31:16 quickfolders-util.js:1292:13
QuickFolders 14:31:16.203  [2 ms]   
 QuickFolders.quickMove.update()
isActive = false quickfolders-util.js:1292:13
QuickFolders 14:31:16.203  [0 ms]   
 toggleMoveModeSearchBox(false) quickfolders-util.js:1292:13
QuickFolders 14:31:16.204  [1 ms]   
 QuickFolders.quickMove.update()
isActive = false quickfolders-util.js:1292:13
QuickFolders 14:31:16.205  [1 ms]   
 toggleMoveModeSearchBox(false) quickfolders-util.js:1292:13
Layout-Darstellung wurde erzwungen, bevor die Seite vollständig geladen war. Falls Stylesheet noch nicht geladen sind, kann dies zu einem kurzzeitigen Darstellung des Inhalts ohne Formatierung führen. 6 aboutMessage.js:58
QuickFolders 14:31:17.6  [801 ms]   
 util.touch(Ärzte)
1692621077
21/8 14:31:17 quickfolders-util.js:1292:13

**Comment by PotatoCarl** [comment]
Any news why the move and filter dialog popup does not work anymore? It works sometimes, but most of the times I do not get the filter dialog. I can move the emails fine, but no filter pops up...  I am currently on quickFilter 6.2.1 and quickFolder 6.1.2.pre24
However, *it looks* like when I start thunderbird the "learning mode" is inactive, despite having "quickFilter assistant" activated. When I turn off the assistant and turn it an again manually, the "learning mode" is active and displayed as red envelope icon... Maybe that is the reason somehow?
Because in that case, the move works...

**Comment by RealRaven2000** [comment]
> Any news why the move and filter dialog popup does not work anymore? It works sometimes, but most of the times I do not get the filter dialog. I can move the emails fine, but no filter pops up... I am currently on quickFilter 6.2.1 and quickFolder 6.1.2.pre24 However, _it looks_ like when I start thunderbird the "learning mode" is inactive, despite having "quickFilter assistant" activated. 

Looks like you found a new bug - most people don;t use the auto-start feature, so it wasn't on my radar. Can you move it to a new issue please? It's very possible that 115 broke it on account it being very different to previous versions.

**Comment by RealRaven2000** [comment]
> Looks like you found a new bug - most people don;t use the auto-start feature, so it wasn't on my radar. Can you move it to a new issue please? It's very possible that 115 broke it on account it being very different to previous versions.

Unless this is the only cause for the assistant not working - in which case I can just change the title??

**Comment by PotatoCarl** [comment]
It actually seems the only cause...

**Comment by RealRaven2000** [comment]
> It actually seems the only cause...

Ok, I hope I can fix it from the quickFilters side, but I do have a new QuickFolders release ready to go, so I am going to check on both sides... (I renamed the title of the issue, glad to come to the root of the problem)

**Comment by RealRaven2000** [comment]
I am investigating this now ; there is a global flag in QuickFolders called `QuickFolders.FilterWorker.FilterMode` - once this is set to true (via a QuickFolders function `toggleFilterMode()` - this is when you use the QF interface, e.g. the QuickFolders tools menu to toggle filter assisted mode):

![image](https://github.com/RealRaven2000/quickFilters/assets/6314674/c90bdc67-5b15-4fc8-9af3-b2fd146c2f67)

However when you use the quickFilters interface and call its own `toggle_FilterMode()` function it will try to also flick the switch for QuickFolders:

```js
    // If QuickFolders is installed, we should also invoke its filter mode
    if (window.QuickFolders) {
      let QF = window.QuickFolders,
          QFwork = QF.FilterWorker ? QF.FilterWorker : QF.Filter;
      // we cannot supress the notification from QuickFolders
      // without adding code in it!
      if (QFwork.FilterMode != active) {// prevent recursion!
        QFwork.toggle_FilterMode ?
          QFwork.toggle_FilterMode(active) :
          QFwork.toggleFilterMode(active);  // (active, silent) !!!
      }

      if (!silent)
        removeOldNotification(notifyBox, active, 'quickfolders-filter');
    }
```

my theory is that this may not work if QuickFolders hasn't been loaded on startup and quickFilters was loaded _beforehand_. This is a little hard to test, because I have no influence on the loading order and cannot debug on startup either. But there must be a way to work around this...

**Comment by RealRaven2000** [comment]
Ok, looks like I am lucky already. We can see in the interface that quickFilters has (automatically) toggled its mode, but QuickFolders didn't get the message and still shows the wrong icon in its tools menu:

![image](https://github.com/RealRaven2000/quickFilters/assets/6314674/5c9ce35b-8329-4bc5-8b69-bc88d6506726)

So it seems quickFilters didn't "see" QF and so couldn't set its assistant mode. I think the easiest fix would be for QuickFolders to look at the (autostart) preference of quickFilters in legacy prefs (about:config). It should then check whether quickFIlters is actually running and has set its own FilterMode=true in the meantime, and then set its own FilterMode accordingly...

**Comment by PotatoCarl** [comment]
Sounds like a plan...

**Comment by RealRaven2000** [comment]
Ok, here is a test version with a quickfix. I added the following code to the QuickFolders `initDelayed()` function:

```js
    if (prefs.getFiltersBoolPref("autoStart")) {
      // quickFilters started before QF and automatically started assistant mode
      if (window.quickFilters && !QuickFolders.FilterWorker.FilterMode) {
        if (window.quickFilters.Util.AssistantActive) {
          QuickFolders.FilterWorker.toggle_FilterMode(true);
        }
      }
    }
```

I have to be very careful here because I don't want it to throw an exception in case quickFilters is not  installed or deactivated.
After setting the flag and restarting Tb, it looks like this:

![image](https://github.com/RealRaven2000/quickFilters/assets/6314674/b0021b45-97d2-42c8-9ce2-9cf8427fc8c8)

Here is the test version:

[QuickFolders-mx-6.2pre55.zip](https://github.com/RealRaven2000/quickFilters/files/12462679/QuickFolders-mx-6.2pre55.zip)

-----

(as usual) to test the version above, download the zip file, drag into Thunderbird Add-ons Manager, do not extract contents or if won't install.

**Comment by RealRaven2000** [comment]
I think we need to move this to the QuickFolders issue list, because I had to fix it from that side... I will try to figure out how to do that. or maybe I will just open a new issue over there...

**Comment by RealRaven2000** [comment]
There is still a problem if we disable and then enable quickFilters (from Add-ons Manager) the filter assistant mode at least looks not enabled. It looks sort of the opposite  of the initial situation:

![image](https://github.com/RealRaven2000/QuickFolders/assets/6314674/b0937474-001b-4177-b9fe-788f1ef3fae5)

I also see the following error caused by QuickFolders in console:

![image](https://github.com/RealRaven2000/QuickFolders/assets/6314674/6b8e71e3-d8c7-4b80-aaa3-b50cdb1450bf)

**Comment by PotatoCarl** [comment]
I see the same here: Disabling the QucikFilter in Addons and then turning it back on the filter assistant is off while QuickFolder thinks it is on...

**Comment by RealRaven2000** [comment]
> I see the same here: Disabling the quickFilters in Addons and then turning it back on the filter assistant is off while QuickFolder thinks it is on...

I may have to fix that particular one on the quickFilters side... but I want to deal with the exception above at least here on QuickFolders. Here is a new version:

[QuickFolders-mx-6.2pre60.zip](https://github.com/RealRaven2000/QuickFolders/files/12465957/QuickFolders-mx-6.2pre60.zip)

As regards the weird behavior of the main quickFilters button not updating when we update the Add-on from extension manager I have raised [[issue 210](https://github.com/RealRaven2000/quickFilters/issues/210)] over on the quickFilters tracker.

**Comment by RealRaven2000** [comment]
Removed some unnecessary processing in the tab listener. Hopefully that didn't change any of the other behavior.

[QuickFolders-mx-6.2pre63.zip](https://github.com/RealRaven2000/QuickFolders/files/12466646/QuickFolders-mx-6.2pre63.zip)

and also uploaded a test version of quickFilters over at [https://github.com/RealRaven2000/quickFilters/issues/210]

-----

(as usual) to test the version above, download the zip file, drag into Thunderbird Add-ons Manager, do not extract contents or if won't install.

**Comment by RealRaven2000** [comment]
Fixed in 6.2 - Published 30/08/2023
